django_lightweight_queue/machine_types.py

```diff
@@ -0,0 +1,101 @@
+from django.utils.functional import cached_property
+
+from . import app_settings
+from .cron_scheduler import CRON_QUEUE_NAME
+
+
+class Machine(object):
+    """
+    Dummy machine class to contain documentation.
+
+    Implementations may extend this class if desired, though this
+    is not required.
+    """
+
+    @property
+    def run_cron(self):
+        """
+        Returns a `bool` for whether or not a runner on this machine should
+        run the cron queue.
+        """
+        raise NotImplementedError()
+
+    @property
+    def worker_names(self):
+        """
+        Returns a list of tuples of (queue_name, worker_num) for the workers
+        which should run on this machine. Worker numbers start at 1.
+
+        Implemetations should be efficient even if this is called several times.
+        """
+        raise NotImplementedError()
+
+
+def get_workers_names(machine_number, machine_count, only_queue):
+    worker_names = []
+
+    # Used to determine the parallelism split
+    job_number = 1
+
+    for queue, num_workers in sorted(app_settings.WORKERS.iteritems()):
+        if only_queue and only_queue != queue:
+            continue
+
+        for worker_num in range(1, num_workers + 1):
+            # We don't go out of our way to start workers on startup - we let
+            # the "restart if they aren't already running" machinery do its
+            # job.
```

Comment:
This comment is somewhat out of context -- we're not doing anything to do with starting workers here (as the function is now called `get_workers_names`). I'd just remove it.